Makefile

```diff
@@ -34,8 +34,36 @@ ct-tls:
 		--suite eredis_tls_SUITE || { docker logs redis; exit 1; }
 	@docker rm -f redis
 
+# Generate and patch documentation.
+# The patching is needed to be able to generate documentation via Elixirs mix.
+# Following changes are needed:
+# - Handle links in headers, change '### <a name="link">Header</a> ###' to
+#                                   '<a name="link"></a> ### Header ###'
+# - Newline needed for before following tags:
+#   </table> </dd> </pre>
+# - Removal of unneeded line breaks (visual only)
+#
+# Note: sed on macOS requires explicit in-place extensions (-i <extension>)
 edoc:
 	@rebar3 edoc skip_deps=true
+	for file in doc/*.md ; do \
+		sed -i.bak 's|### <a name="\(.*\)">\(.*\)</a> ###|<a name="\1"></a>\n### \2 ###|g' $${file} ; \
+		sed -i.bak 's|</table>|\n</table>|g' $${file} ; \
+		sed -i.bak 's|</dd>|\n</dd>|g' $${file} ; \
+		sed -i.bak 's|</code></pre>|</code>\n</pre>|g' $${file} ; \
+		sed -i.bak 's|<br />||g' $${file} ; \
+		rm $${file}.bak ; \
+	done
+
+publish: edoc
+	@touch doc/.build # Prohibit ex_doc to remove .md files
```

Comment:
This .build file was a sneaky thing. I was not able to generate anything today (just like when I started with doc generating with mix). The reason was I started with a clean repo.
If this file is not found ex_doc seems to first remove the doc/*.md files and then trying to parse them, which naturally fails..really weird.

When I first started with mix I had the same problem, but now I understand that when I remove the specified .md-files in `extras` in `mix.exs` it triggered the ex_doc to create the .build file for the first time. Then it worked when reverting the mix.exs change..

Reply:
I also had this problem. Lost a morning in this 😂
But didn't understood why...thanks for the clarification!